pkg/arvo/sys/zuse.hoon

```diff
@@ -5453,6 +5453,35 @@
     |=  cis/coin
     ?.  ?=({$$ case} cis)  ~
     `(unit beam)`[~ [who dex `case`p.cis] (flop t.t.t.pax)]
+  ::
+  ++  json-rn                                           ::  json to rn parser
+    ;~  plug
+      (easy %d)
+      ;~(pose (cold | hep) (easy &))
+      ;~  plug  dim:ag
+        ;~  pose
+          ;~  pfix  dot
+            %+  cook
+            |=(a=tape :-((lent a) (scan a dum:ag)))
+            (plus (shim '0' '9'))
```

Comment:
A parser shouldn't crash, so this should be `+sear` and `+rust` instead of `+cook` and `+scan`. And `%+` should use backstep indentation:

```
%+  sear
  |=(a=tape [(lent a) (rust a dum:ag)])
(plus (shim '0' '9'))
```

Reply:
I'm getting %redo-match with your suggestion. Fixed the backstep indentation and dumb `:-` usage though.

My `scan` was inspired by `++royl-rn` in hoon.hoon
```
=/  moo
  |=  a=tape
  :-  (lent a)
  (scan a (bass 10 (plus sid:ab)))
```

Reply:
Ah, that's 'cause my suggestion was dumb ... ;)

It should be this:

```
%+  sear
  |=  a=tape
  =/  b  (rust a dum:ag)
  ?~  b  ~
  (some [(lent a) u.b])
(plus (shim '0' '9'))
```

The pattern in `+royl-rn` is also wrong. Parsers shouldn't crashes, but `+scan` does.

Reply:
Done!

Reply:
I can see the argument, but note that `dum:ag` is basically `(bass 10 (plus (shim '0' '9')))`? So there aren't any actual inputs it crash on.

Ofc this redundancy kinda suggests a parser of
```hoon
=/  bass10  =|(a=@ |=(b=(list @) ?~(b a $(a (add (mul 10 a) i.b), b t.b))))
%+  cook
  |=(a=(list @u) [(lent a) (bass10 a)])
(plus din:ab)
```
for both. (This probably wasn't originally done bc the `bass` code exists as a combinator but not as a plain function. Except `++can` for bex'en I guess)

Alternately you could
```hoon
=/  prelen
  |=  [a=tape b=tape]
  ?: =(a b) 0
  ?~  a  ~|(%b-must-suffix-a !!)
  +($(a t.a))
=/  with-length
  |*  a=rule
  |=  b=nail
  =/  c  (a b)
  ?~  q.c  c
  c(p.q :_(p.q.c (prelen q.b q.q.c)))
::
(with-length dum:ag)
```
but that seems like rather overkill for two uses :)

Reply:
@ohAitch, there are various use-cases in userspace: my DER parser specifically, but really anything that parses to +octs or equivalent. So this would be worth doing at some point.

Reply:
No complaint from me, it's definitely the version that best isolates parser implementation from declarative grammar specification~